runtime/vara/src/integration_tests.rs

```diff
@@ -345,10 +348,239 @@ fn tokens_locking_works() {
                     program_id.into(),
                     vec![],
                     10_000_000_000,
-                    11 * UNITS,
+                    11_000 * UNITS,
                     false,
                 ),
                 pallet_gear_bank::Error::<Runtime>::InsufficientBalance
             );
         });
 }
+
+#[test]
+fn treasury_surplus_is_not_burned() {
+    init_logger();
+
+    let alice = AccountKeyring::Alice;
+    let bob = AccountKeyring::Bob;
+    let charlie = AccountKeyring::Charlie;
+    let dave = AccountKeyring::Dave;
+
+    let treasury_id = Treasury::account_id();
+
+    ExtBuilder::default()
+        .initial_authorities(vec![
+            (
+                alice.into(),
+                charlie.into(),
+                alice.public(),
+                ed25519::Pair::from_string("//Alice", None)
+                    .unwrap()
+                    .public(),
+                alice.public(),
+                alice.public(),
+            ),
+            (
+                bob.into(),
+                dave.into(),
+                bob.public(),
+                ed25519::Pair::from_string("//Bob", None).unwrap().public(),
+                bob.public(),
+                bob.public(),
+            ),
+        ])
+        .stash(STASH)
+        .endowment(ENDOWMENT)
+        .endowed_accounts(vec![charlie.into(), dave.into()])
+        .root(alice.into())
+        .build()
+        .execute_with(|| {
+            // Treasury pot is empty in the beginning
+            assert_eq!(Treasury::pot(), 0);
+
+            let initial_total_issuance = Balances::total_issuance();
+
+            // Top up treasury balance
+            assert_ok!(Balances::transfer(
+                RuntimeOrigin::signed(charlie.to_account_id()),
+                sp_runtime::MultiAddress::Id(treasury_id.clone()),
+                1_000 * UNITS,
+            ));
+            assert_eq!(Treasury::pot(), 1_000 * UNITS);
+
+            // Run chain for a day so that `Treasury::spend_funds()` is triggered
+            run_to_block(DAYS);
+
+            // Treasury had a surplus, but none of it was burned
+            assert_eq!(Treasury::pot(), 1_000 * UNITS);
+
+            // The total issuance persisted
+            assert_eq!(Balances::total_issuance(), initial_total_issuance);
+
+            // Run chain until another `Treasury::spend_funds()` invocation
```

Comment:
do we have some event on spend_funds call to assert that it was triggered?

Reply:
The `Treasury` pallet emits the events every `T::SpendPeriod` blocks (once a day in out runtime)